src/routes/formsg/formsgSiteLaunch.ts

```diff
@@ -350,6 +356,35 @@ export class FormsgSiteLaunchRouter {
               `Unable to get dig response for domain: ${launchResult.value.primaryDomainSource}. Skipping check for AAAA records`
             )
           }
+
+          if (!caaDigResponse) {
+            logger.info(
+              `Unable to get dig response for domain: ${launchResult.value.primaryDomainSource}. Skipping check for CAA records`
+            )
+          } else if (caaDigResponse.answer) {
+            const caaRecords = caaDigResponse.answer
+
+            /**
+             * NOTE: If there exists more than one CAA Record, we need to
+             * 1. check if they have whitelisted Amazon CAA
+             * 2. if not, send email to inform them to whitelist Amazon CAA
+             */
+            const hasAmazonCAAWhitelisted = caaRecords.some((record) => {
```

Comment:
the assumption we're making here is that if there's only one CAA record, it's the whitelisted amazon one right?

Reply:
not really, the condition to check here is: if it has any CAA record, at least one has to be the amazon one
in the case where there is no CAA record, actually all this is ok already